src/olympia/editors/helpers.py

```diff
@@ -578,7 +579,9 @@ def get_actions(self, request, addon):
                                      'label': label}
             # An unlisted sideload add-on, which requests a full review, cannot
             # be granted a preliminary review.
-            if addon.is_listed or self.review_type == 'preliminary':
+            prelim_allowed = not waffle.flag_is_active(
+                request, 'no-prelim-review') and addon.is_listed
```

Comment:
nit: doing `addon.is_listed` first saves a query for the waffle if the add-on is unlisted.